File: packages/javascript/lib/node/os/getCpuCount.ts
import { availableParallelism } from 'node:os';

export function getCpuCount(): number {
	// The cores this process may actually use, which a CPU affinity mask narrows.
	// Counting the entries of `cpus()` would report every core on the machine.
	return availableParallelism() || 1;
}

File: packages/python/qsu/os/getCpuCount.py
import os


def getCpuCount() -> int:
	# The cores this process may actually use, which a CPU affinity mask narrows.
	# Counting every core on the machine would ignore that.
	if hasattr(os, 'sched_getaffinity'):
		return len(os.sched_getaffinity(0)) or 1

	return os.cpu_count() or 1
